Add MH110 support (predecessor of GP110 with DX200 controller)

cjue opened the pull request:
As far as the datasheets tell MH110 and GP110 are identical except for some kg of total weight and DX200 vs YRC1000 controller.

The URDF references the GP110 meshes for the collision models, and package.xml lists motoman_gp110_support as a dependency.

@EricMarcil Could you please check if the joint effort limits are the same for GP110 and MH110? 

I couldn't find any 3D CAD model for the MH110, but it appears to be identical to the GP110.

This MR build upon #476

EricMarcil commented:
@cjue The robot 3D model are identical.  There is a slight different for the effort on some axis.  See attached file: 
[Robot_Specs_MH110-A0.csv](https://github.com/ros-industrial/motoman/files/8315394/Robot_Specs_MH110-A0.csv)

cjue commented:
Thanks, I updated all the values. The support package should be complete then.

gavanderhoorn commented on motoman_mh110_support/launch/test_mh110.launch:
```diff
@@ -0,0 +1,9 @@
+<launch>
+  <include file="$(find motoman_mh110_support)/launch/load_mh110.launch" />
+
+  <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" />
+
+  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
+
+  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find industrial_robot_client)/config/robot_state_visualize.rviz" required="true" />
```
Please remove empty lines.

gavanderhoorn commented on motoman_mh110_support/package.xml:
```diff
@@ -0,0 +1,55 @@
+    </ul>
+    <p>
+      Joint limits and maximum joint velocities are based on the information 
+      found in the online documentation files for the mostly identical successor GP110 model from
+      https://www.yaskawa.eu.com/products/robots/handling-mounting/productdetail/product/gp110_704 )
```
Same question/suggestion here as in #476.

gavanderhoorn commented on motoman_mh110_support/test/launch_test.xml:
```diff
@@ -0,0 +1,51 @@
+<!--
```
Please also rename this file to include the variant name.

gavanderhoorn commented on motoman_mh110_support/urdf/mh110_macro.xacro:
```diff
@@ -0,0 +1,169 @@
+    <link name="${prefix}base_link">
+      <visual>
+        <geometry>
+          <mesh filename="package://motoman_gp110_support/meshes/gp110/visual/base_link.stl"/>
```
Could you please add a comment here explaining why the MH110 package is using files from the GP110 package?

Just one here should suffice. No need to repeat it on the other `mesh` elements.

gavanderhoorn commented on motoman_mh110_support/urdf/mh110_macro.xacro:
```diff
@@ -0,0 +1,169 @@
+    <!-- ROS-Industrial 'flange' frame: attachment point for EEF models -->
+    <link name="${prefix}flange"/>
+    <joint name="${prefix}joint_6_t-flange" type="fixed">
+      <origin xyz="0.200 0 0" rpy="${radians(180)} ${radians(-90)} 0"/>
```
Please check whether the rotation here should be moved to the `tool0` joint instead.

gavanderhoorn commented on motoman_mh110_support/package.xml:
```diff
@@ -0,0 +1,55 @@
+
+  <exec_depend>joint_state_publisher_gui</exec_depend>
+  <exec_depend>motoman_driver</exec_depend>
+  <exec_depend>motoman_gp110_support</exec_depend>
```
Could you add a comment here explaining why this `exec_depend` exists?

gavanderhoorn reviewed (changes requested):
Some in-line comments.

gavanderhoorn commented:
Just realised the squash-merge in #476 made merging this a bit more difficult.

@cjue: you should be able to drop just about every commit here, and squash e092b34 & fbc8857 together.

cjue commented:
Rebased onto kinetic-devel after #476 merge.

I updated the package.xml regarding the differences of MH110/GP110. 
Unfortunately I couldn't find any official Yaskawa URLs or downloads for the MH110.

gavanderhoorn commented:
> Unfortunately I couldn't find any official Yaskawa URLs or downloads for the MH110.

If you search for `motoman MH110`, you'll end up on the [Yaskawa Motoman product documentation](https://www.motoman.com/en-us/service-training/product-documentation) site. That lists `HW1482741` as the MOTOMAN-MH110 INSTRUCTIONS manual (`169859-1CD`), which lists the *Basic Specifications* of this particular robot in section `5.1`:

![image](https://user-images.githubusercontent.com/4550046/159511200-3e7f6eaf-ad1b-4a2b-904c-ac6ed4c1b503.png)

gavanderhoorn commented on motoman_mh110_support/urdf/mh110_macro.xacro:
```diff
@@ -0,0 +1,170 @@
+      <child link="${prefix}link_3_u"/>
+      <origin xyz="0 0 0.870" rpy="0 0 0" />
+      <axis xyz="0 -1 0" />
+      <limit lower="${radians(-80)}" upper="${radians(90)}" effort="4167.83" velocity="${radians(130)}"/>
```
The limits in `169859-1CD` (see https://github.com/ros-industrial/motoman/pull/477#issuecomment-1075285083) are different: `[-185; 160]`.

@EricMarcil: would you know what causes the difference?

gavanderhoorn replied:
Oh, is that the "limit wrt ground plane vs actual joint limits"?

cjue replied:
Thanks for the link. 

The specifications table uses a weird definition for the U-Axis limits, somehow combining the valid range of motion for the L and U Axis.

EricMarcil replied:
Yes, that would be the case.  The manual limits make reference to the upper arm position relative to the ground which is actually a combination of both L and U axes.  So not useful for the URDF model...

cjue replied:
This came up before in discussion of the MH50: https://github.com/ros-industrial/motoman/issues/282#issuecomment-510516570

Looks to me as if the MH-series documentation generally used that notation.

gavanderhoorn commented on motoman_mh110_support/package.xml:
```diff
@@ -0,0 +1,64 @@
+      Visual appearance, joint limits and maximum joint velocities are 
+      identical to the successor GP110 model, the only differences being
+      joint effort limits and DX200 vs YRC1000 controller connection.
+      Information on the MH110 can be found in <em>HW1482741 (169859-1CD Rev. 2)</em>
+      at the <a href="https://www.motoman.com/en-us/service-training/product-documentation">Yaskawa Motoman product documentation site</a>.
```
```suggestion
      Information on the MH110 can be found in <em>HW1482741 (169859-1CD Rev. 2)</em>.
```

gavanderhoorn replied:
It's clear this is a reference to a Yaskawa Motoman released document. That should be sufficient to find it again if needed.

gavanderhoorn reviewed (approved):
This looks OK now I believe.

Further fixups/corrections should be submitted in follow-up PRs.

gavanderhoorn commented:
Using squash-merge again here to avoid fixups getting into the repository history.

gavanderhoorn commented:
Thanks @cjue :+1:
